File: src/design_patterns/creational/builder/php/pizza/Waiter.php
<?php

class Waiter {

	protected $pizzaBuilder;

	/**
	 * @param PizzaBuilder $pizzaBuilder
	 */
	public function setPizzaBuilder(PizzaBuilder $pizzaBuilder) {
		$this->pizzaBuilder = $pizzaBuilder;
	}

	/**
	 * @return mixed
	 */
	public function getPizza() {
		return $this->pizzaBuilder->getPizza();
	}

	public function constructPizza() {
		$this->pizzaBuilder->createNewPizzaProduct();
		$this->pizzaBuilder->buildDough();
		$this->pizzaBuilder->buildSauce();
		$this->pizzaBuilder->buildTopping();
	}
}

File: src/design_patterns/creational/builder/java/Pizza/Waiter.java
/* "Director" */
class Waiter {
	private PizzaBuilder pizzaBuilder;

	public void setPizzaBuilder(PizzaBuilder pb) { pizzaBuilder = pb; }

	public Pizza getPizza() { return pizzaBuilder.getPizza(); }

	public void constructPizza() {
		pizzaBuilder.createNewPizzaProduct();
		pizzaBuilder.buildDough();
		pizzaBuilder.buildSauce();
		pizzaBuilder.buildTopping();
	}
}

